### ISSUE TITLE
Line edits in settings config dialog resets cursor position on save.

### ISSUE THREAD

**Issue by spongeboburu** [godotengine/godot#20708]
**Godot version:**
3.1-dev

**OS/device including version:**
Fedora 28 64-bit

**Issue description:**
When typing into a line edit in the settings editor (such as "exec_flags" for external editor) and pause, the settings config dialog will call _settings_save and rebuild the UI causing the cursor to start at position 0. So, either type your thing very fast or prepare to be very annoyed.

**Steps to reproduce:**
Editor Settings -> Text Editor -> External type into exec_flags and wait for 1.5s.

** More info **
In 3.0.6 this is not the behaviour, perhaps because it only saves on close instead of on the 1.5 timeout? I much prefer save on close anyway, but if you have save every 1.5 seconds it's not a very good idea to rebuild the UI tree from scratch and updating it because it resets user interaction with the controls (such as cursor position while editing line text).

I also saw there was commented code for "ok" and "cancel" buttons, that sounds like a good idea to have for the settings.

**Comment by elijahscherz** [comment]
I was able to also see this in the Windows version as well and do agree it is a bit annoying.

**I would like to hop on this issue,** but want to ask - do we think a good approach to this is moving back to a save on close ("ok" / "cancel"), try and resolve the cursor resetting issue when the UI is reloaded, or simply increase the timeout allowing for a bit more time to type.

**Comment by Calinou** [comment]
@elijahscherz I would prefer if we kept the current "automatic saving" behavior. Otherwise, people may be confused why their settings don't apply after changing them (especially when following old tutorials).

**Comment by elijahscherz** [comment]
@Calinou I agree I think it can be helpful. It seems to me that the best solution would be to have it still automatically save as it does now, but maybe we can delay the UI refresh (until they leave box) or maintain the cursor position despite a refresh.

**Comment by KoBeWi** [comment]
Same underlying issue as #23626

**Comment by HolonProduction** [comment]
#58783 somewhat fixed this. Only setting were it still occurs is `interface/inspector/resources_to_open_in_new_inspector`, might make sense to not rebuild the inspector if this setting changes since it is only relevant for opening an inspector.

**Comment by KoBeWi** [comment]
Everything in `inspector` category causes inspector update.

**Comment by HolonProduction** [comment]
Yes but the rebuilding is only really a problem for line edits. And everything else in this section are other controls.
